pkg/common/db/db.go

```diff
@@ -0,0 +1,68 @@
+package db
+
+import (
+	"fmt"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"log"
+	"os"
+	"os/exec"
+	"time"
+)
+
+// Init is used to establish connection with database.
+func Init(url string) *gorm.DB {
+	db, err := gorm.Open(postgres.Open(url), &gorm.Config{})
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	waitDbIsUp(db)
+
+	return db
+}
+
+// waitDbIsUp is a function that prevents application container to fail if docker-compose is used and db hasn't started yet.
+func waitDbIsUp(db *gorm.DB) {
```

Comment:
I think that this functional can be delegated to external instrument like docker-compose. See this [example](https://github.com/peter-evans/docker-compose-healthcheck)